src/modules/ods/ods.controller.test.ts

```diff
@@ -1,30 +1,74 @@
+import { CUSTOMERS, DEALS } from '@ukef/constants';
 import { when } from 'jest-when';
-import { CUSTOMERS } from '@ukef/constants';
+
 import { OdsController } from './ods.controller';
 import { OdsService } from './ods.service';
 
 describe('OdsController', () => {
   let odsServiceFindCustomer: jest.Mock;
+  let odsServiceFindDeal: jest.Mock;
 
   let controller: OdsController;
 
   beforeEach(() => {
     const odsService = new OdsService(null, null);
+
     odsServiceFindCustomer = jest.fn();
     odsService.findCustomer = odsServiceFindCustomer;
 
+    odsServiceFindDeal = jest.fn();
+    odsService.findDeal = odsServiceFindDeal;
+
     controller = new OdsController(odsService);
   });
 
   describe('findCustomer', () => {
-    it('should return the customer when a valid customer URN is provided', async () => {
-      const mockCustomerDetails = { urn: CUSTOMERS.EXAMPLES.PARTYURN, name: 'Test name' };
+    it('should call odsService.findCustomer', async () => {
+      // Act
+      await controller.findCustomer({ urn: CUSTOMERS.EXAMPLES.PARTYURN });
+
+      // Assert
+      expect(odsServiceFindCustomer).toHaveBeenCalledTimes(1);
+
+      expect(odsServiceFindCustomer).toHaveBeenCalledWith(CUSTOMERS.EXAMPLES.PARTYURN);
+    });
+
+    it('should return a customer when a valid customer URN is provided', async () => {
+      // Arrange
+      const mockCustomerDetails = { urn: CUSTOMERS.EXAMPLES.PARTYURN, name: 'Test customer name' };
 
       when(odsServiceFindCustomer).calledWith(CUSTOMERS.EXAMPLES.PARTYURN).mockResolvedValueOnce(mockCustomerDetails);
 
-      const customers = await controller.findCustomer({ urn: CUSTOMERS.EXAMPLES.PARTYURN });
+      // Act
+      const result = await controller.findCustomer({ urn: CUSTOMERS.EXAMPLES.PARTYURN });
+
+      // Assert
+      expect(result).toEqual(mockCustomerDetails);
+    });
+  });
+
+  describe('findDeal', () => {
+    it('should call odsService.findDeal', async () => {
```

Comment:
Can we add some assertions for both the test cases where an exception or non `200` response is received please,.